sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -83,6 +89,284 @@ protected void afterTest() {
+            })
+            .verifyComplete();
+        assertNotNull(transaction.get());
+
+        // Assert & Act
+        StepVerifier.create(sender.send(message2, transaction.get()))
+            .verifyComplete();
+
+        final ServiceBusReceivedMessageContext receivedContext = receiver.receive().next().block(TIMEOUT);
+        assertNotNull(receivedContext);
+
+        final ServiceBusReceivedMessage receivedMessage = receivedContext.getMessage();
+        assertNotNull(receivedMessage);
+        final Mono<Void> operation;
+        if (DispositionStatus.ABANDONED == dispositionStatus && isSessionEnabled) {
+            operation = receiver.abandon(receivedMessage, null, sessionId, transaction.get());
+        } else if (DispositionStatus.ABANDONED == dispositionStatus && !isSessionEnabled) {
+            operation = receiver.abandon(receivedMessage, null, transaction.get());
+        } else if (DispositionStatus.SUSPENDED == dispositionStatus && isSessionEnabled) {
+            DeadLetterOptions deadLetterOptions = new DeadLetterOptions().setDeadLetterReason(deadLetterReason);
+            operation = receiver.deadLetter(receivedMessage, deadLetterOptions, sessionId, transaction.get());
+        } else if (DispositionStatus.SUSPENDED == dispositionStatus && !isSessionEnabled) {
+            DeadLetterOptions deadLetterOptions = new DeadLetterOptions().setDeadLetterReason(deadLetterReason);
+            operation = receiver.deadLetter(receivedMessage, deadLetterOptions, transaction.get());
+        } else if (DispositionStatus.COMPLETED == dispositionStatus && isSessionEnabled) {
+            operation = receiver.complete(receivedMessage, sessionId, transaction.get());
+        } else if (DispositionStatus.COMPLETED == dispositionStatus && !isSessionEnabled) {
+            operation = receiver.complete(receivedMessage, transaction.get());
+        } else if (DispositionStatus.DEFERRED == dispositionStatus && isSessionEnabled) {
+            operation = receiver.defer(receivedMessage, null, sessionId, transaction.get());
+        } else if (DispositionStatus.DEFERRED == dispositionStatus && !isSessionEnabled) {
+            operation = receiver.defer(receivedMessage, null, transaction.get());
+        } else {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                "Disposition status not recognized for this test case: " + dispositionStatus));
+        }
+
+        StepVerifier.create(operation)
+            .verifyComplete();
+
+        if (commitTransaction) {
+            StepVerifier.create(receiver.commitTransaction(transaction.get()))
+                .verifyComplete();
+        } else {
+            StepVerifier.create(receiver.rollbackTransaction(transaction.get()))
+                .verifyComplete();
+        }
+    }
+
+    /**
+     * Verifies that we can do following on different clients i.e. sender and receiver.
+     * 1. create transaction using sender
```

Comment:
There is only one operation in here. I'm not sure this is actually testing a transaction.

Reply:
It overlaps functionality of the previous test method. I think we can just remove this.